### ISSUE TITLE
TextBlock.TextWrapping="Wrap" doesn't wrap and changes the layout of my window

### ISSUE THREAD

**Issue by derekantrican** [AvaloniaUI/Avalonia#4838]
Avalonia window without the TextBlock at the top:
<img src="https://user-images.githubusercontent.com/1558019/95667814-260a4b80-0b20-11eb-9762-2304e8984aa6.png" width="40%" height="40%">

Avalonia window *with* the TextBlock (note the result is the same if I set `TextWrapping="Wrap"` on the TextBlock):
<img src="https://user-images.githubusercontent.com/1558019/95667825-45a17400-0b20-11eb-8c5a-08ebafc9e10d.png" width="40%" height="40%">

Same code in WPF without `TextWrapping="Wrap"`:
<img src="https://user-images.githubusercontent.com/1558019/95667843-7aadc680-0b20-11eb-8833-b4a05aa81fca.png" width="40%" height="40%">

Same code in WPF *with* `TextWrapping="Wrap"`:
<img src="https://user-images.githubusercontent.com/1558019/95667851-91ecb400-0b20-11eb-85f7-783bd6e5bb8b.png" width="40%" height="40%">

Notice how WPF doesn't shift the contents outside of the bounds of the window - regardless of whether I have set `TextWrapping="Wrap"` or not.

Here is my code:

```xml
<Window x:Class="MyProjectWPF.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:MyProjectWPF"
        mc:Ignorable="d"
        Title="MainWindow" MaxWidth="300" Height="420">
    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="*"/>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="Auto"/>
        </Grid.RowDefinitions>
        <TextBlock Text="Describe the bug and what you were doing when you encountered it" TextWrapping="Wrap" Margin="5"/>
        <TextBox Grid.Row="1" Width="NaN" Height="200" Text="{Binding Description}" 
             TextWrapping="Wrap" AcceptsReturn="True" Margin="5"/>
        <DockPanel Grid.Row="2">
            <TextBlock Text="Your name:" Margin="5"/>
            <TextBox Text="{Binding Name}" Margin="5"/>
        </DockPanel>
        <DockPanel Grid.Row="3">
            <TextBlock Text="Email:" Margin="5"/>
            <TextBox Text="{Binding Email}" Width="NaN" Margin="5"/>
        </DockPanel>
        <StackPanel Grid.Row="4">
            <RadioButton Content="Bug Report" Margin="5"/>
            <RadioButton Content="Feature Request" Margin="5"/>
        </StackPanel>
        <TextBlock Grid.Row="5" HorizontalAlignment="Left" Text="--LOG FILES ATTACHED--" 
               Background="Green" FontWeight="Bold" Margin="5"/>
        <Button Grid.Row="5" HorizontalAlignment="Right" Content="Submit" Command="{Binding Submit}" Margin="5"/>
    </Grid>
</Window>
```

**Comment by FoggyFinder** [comment]
Set `ColumnDefinitions` to `*`

```xml
<Grid ColumnDefinitions="*">
...
</Grid>
```

**Comment by derekantrican** [comment]
@FoggyFinder Unfortunately that doesn't help at all

**Comment by FoggyFinder** [comment]
@derekantrican what version do you use?

**Comment by derekantrican** [comment]
Avalonia 0.9.12

**Comment by FoggyFinder** [comment]
Odd. It works as expected for me on Windows even without explicit ColumnDefinitions="*"

Can you attach MCVE?

**Comment by derekantrican** [comment]
Here you go. This took a bit to track down and I found that I had trouble reproducing it with just a MainWindow alone - it seems to be dependent on being shown (as a dialog, specifically) from another window.

I also tried my MCVE with `<Grid ColumnDefinitions="*">` (with no luck) and I tried my MCVE in vanilla WPF with no issues.

Here are the important bits:

### MainWindow.axaml: *(the UI here is not so important, I think. Just need some way to invoke the event in the code-behind)*
```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" Width="100" Height="50"
        x:Class="AvaloniaTest.MainWindow"
        Title="MainWindow">
  <Button Content="Click Me" Click="OpenSubmitReport" Margin="5"/>
</Window>
```

### MainWindow.xaml.cs:
```csharp
using Avalonia;
using Avalonia.Controls;
using Avalonia.Interactivity;
using Avalonia.Markup.Xaml;

namespace AvaloniaTest
{
    public class MainWindow : Window
    {
        public MainWindow()
        {
            this.InitializeComponent();
#if DEBUG
            this.AttachDevTools();
#endif
        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);
        }

        public void OpenSubmitReport(object sender, RoutedEventArgs args)
        {
            SubmitReport submitReport = new SubmitReport();
            submitReport.ShowDialog(this); //ShowDialog seems to be critical - does not reproduce with "Show"
        }
    }
}
```

### SubmitReport.axaml: *(this could be probably trimmed down some for a MCVE but I wanted to keep it similar to my screenshots above)*
```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" MaxWidth="300" Height="420"
        x:Class="AvaloniaTest.SubmitReport"
        Title="AvaloniaTest">
  <Grid ColumnDefinitions="*">
    <Grid.RowDefinitions>
      <RowDefinition Height="Auto"/>
      <RowDefinition Height="*"/>
      <RowDefinition Height="Auto"/>
      <RowDefinition Height="Auto"/>
      <RowDefinition Height="Auto"/>
      <RowDefinition Height="Auto"/>
    </Grid.RowDefinitions>
    <TextBlock Text="Describe the bug and what you were doing when you encountered it" TextWrapping="Wrap" Margin="5"/>
    <TextBox Grid.Row="1" Width="NaN" Height="200"
         TextWrapping="Wrap" AcceptsReturn="True" Margin="5"/>
    <DockPanel Grid.Row="2">
      <TextBlock Text="Your name:" Margin="5"/>
      <TextBox Margin="5"/>
    </DockPanel>
    <DockPanel Grid.Row="3">
      <TextBlock Text="Email:" Margin="5"/>
      <TextBox Width="NaN" Margin="5"/>
    </DockPanel>
    <StackPanel Grid.Row="4">
      <RadioButton Content="Bug Report" Margin="5"/>
      <RadioButton Content="Feature Request" Margin="5"/>
    </StackPanel>
    <TextBlock Grid.Row="5" HorizontalAlignment="Left" Text="--LOG FILES ATTACHED--"
           Background="Green" FontWeight="Bold" Margin="5"/>
    <Button Grid.Row="5" HorizontalAlignment="Right" Content="Submit" Margin="5"/>
  </Grid>
</Window>
```

**Comment by derekantrican** [comment]
@FoggyFinder any new ideas? Are you able to reproduce now?

**Comment by FoggyFinder** [comment]
@derekantrican nope, still works as expected. Platform?

**Comment by maxkatz6** [comment]
I don't remember, when this issue was fixed. But I can reproduce it with 0.9.12 and cannot with 0.10.0.

With 0.9.12
![image](https://user-images.githubusercontent.com/3163374/96329728-bc0d0d00-101d-11eb-90e6-435f977b23b9.png)


With 0.10.0-preview5
![image](https://user-images.githubusercontent.com/3163374/96329711-9849c700-101d-11eb-81b0-58f53357fd15.png)

**Comment by maxkatz6** [comment]
As a workaround for 0.9, instead of 
```<Grid ColumnDefinitions="*">```
You can put binding there
```<Grid Width="{Binding $parent[Window].Bounds.Width}">```

**Comment by derekantrican** [comment]
Yes, that workaround of `<Grid Width="{Binding $parent[Window].Bounds.Width}">` seems to solve it!

**Comment by 0x90d** [comment]
@maxkatz6 I'm using `0.10.999-cibuild0014036-beta` and I have the same bug (Windows only!), your workaround still works. Unfortunately the nightly publish is broken so I dont know if this is fixed and my version is older than `0.10.0-preview5`, or if this is broken again.

**Comment by 0x90d** [comment]
FYI: This workaround fixes the problem on windows but causes a bug under Linux making the Window always being large. A screenshot is available at https://github.com/0x90d/videoduplicatefinder/issues/212#issue-1007183950
